fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java

```diff
@@ -1271,65 +1277,133 @@ private static ColumnDef findPartitionDefByName(List<ColumnDef> columnDefs, Stri
+                    slot = (SlotRef) slot.clone();
+                    slot.setTblName(node.getName());
+                    return slot;
+                }
+                @Override
+                public Expr visitView(ViewRelation node, SlotRef slot) {
+                    TableName tableName = slot.getTblNameWithoutAnalyzed();
+                    if (tableName != null && !node.getResolveTableName().equals(tableName)) {
+                        return null;
+                    }
+                    slot = (SlotRef) slot.clone();
+                    slot.setTblName(null); //clear table name here, not check it inside
+                    return node.getQueryStatement().getQueryRelation().accept(this, slot);
+                }
+                @Override
+                public Expr visitJoin(JoinRelation node, SlotRef slot) {
+                    Relation leftRelation = node.getLeft();
+                    Expr leftExpr = leftRelation.accept(this, slot);
+                    if (leftExpr != null) {
+                        return leftExpr;
+                    }
+                    Relation rightRelation = node.getRight();
+                    Expr rightExpr = rightRelation.accept(this, slot);
+                    if (rightExpr != null) {
+                        return rightExpr;
+                    }
+                    return null;
                 }
-                return null;
-            }
 
-            @Override
-            public Expr visitSetOp(SetOperationRelation node, Void context) {
-                for (Relation relation : node.getRelations()) {
-                    Expr resolved = relation.accept(this, null);
-                    if (resolved != null) {
-                        return resolved;
+                @Override
+                public Expr visitSetOp(SetOperationRelation node, SlotRef slot) {
+                    for (Relation relation : node.getRelations()) {
+                        Expr resolved = relation.accept(this, slot);
+                        if (resolved != null) {
+                            return resolved;
+                        }
                     }
+                    return null;
                 }
-                return null;
-            }
 
-            @Override
-            public SlotRef visitValues(ValuesRelation node, Void context) {
-                return null;
-            }
-        };
-        return queryStatement.getQueryRelation().accept(slotRefResolver, null);
+                @Override
+                public SlotRef visitValues(ValuesRelation node, SlotRef slot) {
+                    return null;
+                }
+            };
+
+            return visitors;
+        }
+    }
+
+    public static Expr resolveSlotRef(SlotRef slotRef, QueryStatement queryStatement) {
+        AstVisitorFactory.AstVisitors visitors = AstVisitorFactory.createAstVisitors();
+        return queryStatement.getQueryRelation().accept(visitors.slotRefResolver, slotRef);
```

Comment:
why slotref is different from expr?

Reply:
`resolveSlotRef` is an original function which is used elsewhere, so I did not remove it